### DISCUSSION TITLE
Change duration/fade out during crossfade based on meta data

### DISCUSSION THREAD

**Discussion by RM-FM** [savonet/liquidsoap#3576] [General]
Hi,

Is there a way to change the crossfade duration and fade out during `cross` for old/a source based on new/b source metadata? I would like to overwrite `liq_cross_duration` and `liq_fade_out` with smaller values in case the next track is a jingle.

I‘m able to get the metadata in the custom
crossfade function, but I didn‘t manage yet to manipulate the crossfade behavior.

Any idea how to achieve that?

**Comment by Moonbase59** [comment]
I’m also looking for this. There should be something analogous to  `new.metadata["jingle_mode"]`. We probably all naively tried to use `old.metadata["liq_cross_duration"] = "0.1"` which doesn’t work.

Would be real nice not having to copy all the crossfading code, just to set one value of metadata "on-the-fly".

**Reply by RM-FM** [reply]
I actually managed to do that. You can use fade in/out delay to move the overlap within the buffer.
